### PR TITLE
GPII-3356 - Move rakefiles from shared/rakefiles to common/rakefiles

### PR DESCRIPTION
**Pull request by amatas**
This PR moves all the files used in the three project types (`aws`, `gcp` and `common`) to a separate directory called `shared`. The `shared` directory contains the `charts` directory, the `rakefiles` used in all the environments and the `version.yml` file generated by the gpii-version-updater.

The https://github.com/gpii-ops/gpii-version-updater/pull/8 must be merged after this PR in order to avoid new `version.yml` files in the wrong location.

### CODE REVIEW

File: common/live/prd/Rakefile [hunk 1]
```diff
@@ -1,5 +1,5 @@
-import "../../../shared/rakefiles/entrypoint.rake"
-import "../../../shared/rakefiles/entrypoint_common.rake"
+import "../../../common/rakefiles/entrypoint.rake"
```
Comment (stepanstipl) [inline common/live/prd/Rakefile]: You can remove `../common` here, no?

File: common/live/prd/Rakefile [hunk 2]
```diff
@@ -1,5 +1,5 @@
...
+import "../../../common/rakefiles/entrypoint_common.rake"
```
Comment (stepanstipl) [inline common/live/prd/Rakefile]: same ^

File: common/live/stg/Rakefile [hunk 3]
```diff
@@ -1,5 +1,5 @@
-import "../../../shared/rakefiles/entrypoint.rake"
-import "../../../shared/rakefiles/entrypoint_common.rake"
+import "../../../common/rakefiles/entrypoint.rake"
+import "../../../common/rakefiles/entrypoint_common.rake"
```
Comment (stepanstipl) [inline common/live/stg/Rakefile]: same ^

File: common/docker-compose.yaml [hunk 4]
```diff
@@ -42,7 +42,7 @@ services:
     volumes:
       - .:/project
       - ../common/charts:/charts
-      - ../shared/rakefiles:/rakefiles
+      - ../common/rakefiles:/rakefiles
```
Comment (stepanstipl) [inline common/docker-compose.yaml]: I think you can remove `../common` from the path here

**Comment by mrtyler** [comment]
Maybe I missed something, but this is not what we agreed on when we (Alfredo and I) last discussed this topic.

I added this link to GPII-3356. It is where the discussion about this topic really began:
https://github.com/gpii-ops/gpii-infra/pull/60#discussion_r205322511

The relative path import problem is a red herring. The reason I vote for having separate `common/` and `shared/` directories is because they contain different things.

`common/` is a self-contained Exekube project. It contains `modules/` and a `live/` hierarchy.

`shared/` is a place for resources used by more than one gpii-infra project. It (should) contain `charts/` and `rakefiles/`.

Combining these two types of data  into a single directory is, for me, confusing and not adding any value. Am I missing something?

**Comment by amatas** [comment]
I added also this comment to GPII-3356:

I'm not able to find a transcription of the discussion about where to store the rakefiles, and the reasons for having the shared directory.

The directory structure defined was:
1. aws: contains only the files directly related to AWS
1. gcp: contains only the files directly related to GCP
1. common: contains only the files directly related to the common resources (DNS, SA, team projects, ...).
1. shared: contains the files that are used in more that one of the previous project types.

I said this because when we need to implement GPII-3359 we need to discard jobs based on the changes made in particular directories. And for me, it seems clearer set a simple condition for the 3 first directories instead setting a condition per subdirectories inside the common directory.

**Comment by natarajaya** [comment]
@mrtyler @amatas I remember our discussions on this topic differently :)
I remember that we all agreed with Tyler's point of `common` being a self-contained exekube project, but we felt okay keeping other common stuff (`charts`) in there too.

Regarding improvements proposed in GPII-3359 – I think we don't want to skip common steps in CI if there is a change in rake files, so if we keep them in `shared` dir, we'll have to have additional check for them anyway.

But I am fine with having another directory for `charts` and `rakefiles` if team decides that it makes project structure less confusing.

**Comment by stepanstipl** [comment]
I'm slightly in favor of distinguishing between `common` and `shared`. `common` being just for common infra part and `shared` for anything shared between `aws`, `gcp` and `common`. In that spirit I would probably move `charts` under `shared` as well (although in long term I think charts should live outside the repo, as those are independent artifacts with their own versioning).

**Comment by natarajaya** [comment]
LGTM

**Comment by amatas** [comment]
> LGTM although you, like Stepan, did not update this section: https://github.com/gpii-ops/gpii-infra/blob/master/aws/README.md#i-want-to-test-my-local-changes-to-gpii-components-in-my-cluster. I guess that's ok since we're tracking that bug in GPII-3384.

Fixed in https://github.com/gpii-ops/gpii-infra/pull/131/commits/87fedbef8c7ee8a84cc33b74617544939637bbc5

**Pull request merged**